pennylane/ops/qubit/matrix_ops.py

```diff
@@ -183,24 +198,47 @@ def __init__(
                 "The control wires must be different from the wires specified to apply the unitary on."
             )
 
-        U = params[0]
-        target_dim = 2 ** len(wires)
-        if len(U) != target_dim:
-            raise ValueError(f"Input unitary must be of shape {(target_dim, target_dim)}")
-
-        # Saving for the circuit drawer
-        self._target_wires = wires
-        self._control_wires = control_wires
-        self.U = U
+        self._hyperparameters = {
+            "u_wires": wires,
+            "control_wires": control_wires,
+            "control_values": control_values,
+        }
```

Comment:
```suggestion
        self.hyperparameters["u_wires"] = wires
        self.hyperparameters["control_wires"] = control_wires
        self.hyperparameters['control_values'] = control_values
```
Another way to potentially do this.